rsocket/framing/ScheduledFrameProcessor.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <folly/io/async/EventBase.h>
+
+#include "rsocket/framing/FrameProcessor.h"
+
+namespace rsocket {
+
+class ScheduledFrameProcessor : public FrameProcessor {
+ public:
+  ScheduledFrameProcessor(
+      std::shared_ptr<FrameProcessor> fp,
```

Comment:
Why not a unique_ptr? I would prefer the object that a class decorates to take the ownership of it.